### ISSUE TITLE
Soft-hyphen character breaks render and movement

### ISSUE THREAD

**Issue by CesarGallego** [helix-editor/helix#11447]
### Summary

Paste text with soft-hypen breaks render, movement and selection when I move thru the pasted text. Whitespace render it's not showing the special character.

As it's discussed on my terminal, this must be a feature of the editor [https://github.com/kovidgoyal/kitty/issues/6978](https://github.com/kovidgoyal/kitty/issues/6978). And there is the same problem on [nvim](https://github.com/neovim/neovim/issues/26945).

Thanks for make this awesome tool, it's my daily driver. Unfortunately I have no idea about Rust to open a pull request.

### Reproduction Steps

 
The asciinema it's not helpfull because they always hide the soft-hypen character, but there it is:
https://asciinema.org/a/3W9QFNz68ZjSosfHEtNpIJzs9
You can see how the theme breaks at the end of the line anyway.

I tried this:

Paste a fingerprint from [ssl-tools.net](https://ssl-tools.net/subjects/badf8ae3f7eb508c94c1bae31e7cdc3a713d4437). When I move around the text mess up and movements behave odd.

I expected this to happen:
I try to see what is happening with set whitespace.render all. To see why all thing it's broken.

Instead, this happened:
Can't move over this part, because movements go crazy with this "empty" space. I can't see this characters with withespace render.

### Helix log

Log it's not helping because it fails silently
```
2024-08-08T11:55:13.087 globset [DEBUG] glob converted to regex: Glob { glob: "*/Dockerfile.*", re: "(?-u)^.*/Dockerfile\\..*$", opts: GlobOptions { case_insensitive: false, literal_separator: false, backslash_escape: true, empty_alternates: false }, tokens: Tokens([ZeroOrMore, Literal('/'), Literal('D'), Literal('o'), Literal('c'), Literal('k'), Literal('e'), Literal('r'), Literal('f'), Literal('i'), Literal('l'), Literal('e'), Literal('.'), ZeroOrMore]) }
2024-08-08T11:55:13.089 globset [DEBUG] glob converted to regex: Glob { glob: "*/dockerfile.*", re: "(?-u)^.*/dockerfile\\..*$", opts: GlobOptions { case_insensitive: false, literal_separator: false, backslash_escape: true, empty_alternates: false }, tokens: Tokens([ZeroOrMore, Literal('/'), Literal('d'), Literal('o'), Literal('c'), Literal('k'), Literal('e'), Literal('r'), Literal('f'), Literal('i'), Literal('l'), Literal('e'), Literal('.'), ZeroOrMore]) }
2024-08-08T11:55:13.089 globset [DEBUG] glob converted to regex: Glob { glob: "*/Containerfile.*", re: "(?-u)^.*/Containerfile\\..*$", opts: GlobOptions { case_insensitive: false, literal_separator: false, backslash_escape: true, empty_alternates: false }, tokens: Tokens([ZeroOrMore, Literal('/'), Literal('C'), Literal('o'), Literal('n'), Literal('t'), Literal('a'), Literal('i'), Literal('n'), Literal('e'), Literal('r'), Literal('f'), Literal('i'), Literal('l'), Literal('e'), Literal('.'), ZeroOrMore]) }
2024-08-08T11:55:13.089 globset [DEBUG] glob converted to regex: Glob { glob: "*/containerfile.*", re: "(?-u)^.*/containerfile\\..*$", opts: GlobOptions { case_insensitive: false, literal_separator: false, backslash_escape: true, empty_alternates: false }, tokens: Tokens([ZeroOrMore, Literal('/'), Literal('c'), Literal('o'), Literal('n'), Literal('t'), Literal('a'), Literal('i'), Literal('n'), Literal('e'), Literal('r'), Literal('f'), Literal('i'), Literal('l'), Literal('e'), Literal('.'), ZeroOrMore]) }
2024-08-08T11:55:13.089 globset [DEBUG] glob converted to regex: Glob { glob: "*/BUILD.*", re: "(?-u)^.*/BUILD\\..*$", opts: GlobOptions { case_insensitive: false, literal_separator: false, backslash_escape: true, empty_alternates: false }, tokens: Tokens([ZeroOrMore, Literal('/'), Literal('B'), Literal('U'), Literal('I'), Literal('L'), Literal('D'), Literal('.'), ZeroOrMore]) }
2024-08-08T11:55:13.089 globset [DEBUG] glob converted to regex: Glob { glob: "*/.env.*", re: "(?-u)^.*/\\.env\\..*$", opts: GlobOptions { case_insensitive: false, literal_separator: false, backslash_escape: true, empty_alternates: false }, tokens: Tokens([ZeroOrMore, Literal('/'), Literal('.'), Literal('e'), Literal('n'), Literal('v'), Literal('.'), ZeroOrMore]) }
2024-08-08T11:55:13.089 globset [DEBUG] glob converted to regex: Glob { glob: "*/.envrc.*", re: "(?-u)^.*/\\.envrc\\..*$", opts: GlobOptions { case_insensitive: false, literal_separator: false, backslash_escape: true, empty_alternates: false }, tokens: Tokens([ZeroOrMore, Literal('/'), Literal('.'), Literal('e'), Literal('n'), Literal('v'), Literal('r'), Literal('c'), Literal('.'), ZeroOrMore]) }
2024-08-08T11:55:13.089 globset [DEBUG] glob converted to regex: Glob { glob: "*/Jenkinsfile.*", re: "(?-u)^.*/Jenkinsfile\\..*$", opts: GlobOptions { case_insensitive: false, literal_separator: false, backslash_escape: true, empty_alternates: false }, tokens: Tokens([ZeroOrMore, Literal('/'), Literal('J'), Literal('e'), Literal('n'), Literal('k'), Literal('i'), Literal('n'), Literal('s'), Literal('f'), Literal('i'), Literal('l'), Literal('e'), Literal('.'), ZeroOrMore]) }
2024-08-08T11:55:13.089 globset [DEBUG] built glob set; 1 literals, 0 basenames, 0 extensions, 0 prefixes, 121 suffixes, 5 required extensions, 8 regexes
2024-08-08T11:55:13.095 helix_view::clipboard [DEBUG] Using pbcopy+pbpaste to interact with the system clipboard
2024-08-08T11:55:13.108 helix_tui::backend::crossterm [DEBUG] The keyboard enhancement protocol is supported in this terminal (checked in 10.6235ms)
2024-08-08T11:55:13.109 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2024-08-08T11:55:13.347 helix_term::application [DEBUG] received editor event: IdleTimer
2024-08-08T11:55:15.123 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2024-08-08T11:55:18.441 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2024-08-08T11:55:22.762 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2024-08-08T11:55:22.763 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 0
2024-08-08T11:55:23.015 helix_term::application [DEBUG] received editor event: IdleTimer
2024-08-08T11:55:24.703 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:24.924 helix_term::application [DEBUG] received editor event: IdleTimer
2024-08-08T11:55:27.256 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:27.507 helix_term::application [DEBUG] received editor event: IdleTimer
2024-08-08T11:55:27.784 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:28.035 helix_term::application [DEBUG] received editor event: IdleTimer
2024-08-08T11:55:28.708 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:28.960 helix_term::application [DEBUG] received editor event: IdleTimer
2024-08-08T11:55:29.209 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:29.292 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:29.376 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:29.459 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:29.543 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:29.627 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:29.709 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:29.794 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:29.878 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:29.962 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:30.045 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:30.129 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:30.213 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:30.298 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:30.381 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:30.463 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:30.548 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:30.633 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:30.715 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:30.798 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:30.883 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:30.967 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:31.050 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:31.135 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:31.218 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:31.303 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:31.387 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:31.469 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:31.557 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:31.638 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:31.720 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:31.804 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:31.887 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:31.971 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:32.057 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:32.139 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:32.223 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:32.307 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:32.391 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:32.473 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:32.559 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:32.642 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:32.725 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:32.810 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:32.893 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:32.977 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:33.062 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:33.313 helix_term::application [DEBUG] received editor event: IdleTimer
2024-08-08T11:55:33.977 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:34.228 helix_term::application [DEBUG] received editor event: IdleTimer
2024-08-08T11:55:35.403 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:35.628 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:35.879 helix_term::application [DEBUG] received editor event: IdleTimer
2024-08-08T11:55:36.096 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:36.219 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 1
2024-08-08T11:55:36.347 helix_term::application [DEBUG] received editor event: IdleTimer
2024-08-08T11:55:36.440 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 2
2024-08-08T11:55:36.691 helix_term::application [DEBUG] received editor event: IdleTimer
2024-08-08T11:55:36.755 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 2
2024-08-08T11:55:37.005 helix_term::application [DEBUG] received editor event: IdleTimer
2024-08-08T11:55:38.464 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 2
2024-08-08T11:55:38.565 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 2
2024-08-08T11:55:38.817 helix_term::application [DEBUG] received editor event: IdleTimer
2024-08-08T11:55:40.469 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 2
2024-08-08T11:55:40.720 helix_term::application [DEBUG] received editor event: IdleTimer
2024-08-08T11:55:41.766 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 2
2024-08-08T11:55:42.017 helix_term::application [DEBUG] received editor event: IdleTimer
2024-08-08T11:55:42.082 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 2
2024-08-08T11:55:42.334 helix_term::application [DEBUG] received editor event: IdleTimer
2024-08-08T11:55:43.643 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 2
2024-08-08T11:55:43.893 helix_term::application [DEBUG] received editor event: IdleTimer
2024-08-08T11:55:43.953 helix_term::commands::typed [DEBUG] quitting...
2024-08-08T11:55:43.954 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 2
2024-08-08T11:55:45.241 helix_view::document [DEBUG] id 1 modified - last saved: 0, current: 2
2024-08-08T11:55:45.572 helix_term::job [DEBUG] waiting on jobs...
2024-08-08T11:55:45.572 helix_term::job [DEBUG] waiting on jobs...
```

### Platform

Linux / MacOS

### Terminal Emulator

kitty / iTerm2

### Installation Method

nixpkgs / brew

### Helix Version

24.3 / 24.7

**Comment by kirawi** [comment]
Duplicate of https://github.com/helix-editor/helix/issues/11255
